src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+
+public class Vision extends SubsystemBase {
+  /** Creates a new ExampleSubsystem. */
+  private NetworkTable table;
+  private NetworkTableEntry tx;
+  private NetworkTableEntry ty;
+  private NetworkTableEntry ta;
+
+  public Vision() {
+   table = NetworkTableInstance.getDefault().getTable("limelight-ups");
+   tx = table.getEntry("tx");
+   ty = table.getEntry("ty");
+   ta = table.getEntry("ta");
+  }
+
+  public void getLimelightValues() {
+    //read values periodically
+    double x = tx.getDouble(0.0);
+    double y = ty.getDouble(0.0);
+    double area = ta.getDouble(0.0);
+
+    //post to smart dashboard periodically
+    SmartDashboard.putNumber("LimelightX", x);
+    SmartDashboard.putNumber("LimelightY", y);
+    SmartDashboard.putNumber("LimelightArea", area);
+  }
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
```

Comment:
Suggest to remove the VisionCommand and instead call getLimelightValues in the periodic method